packages/sitecore-jss-nextjs/src/middleware/editing-data-cache.ts

```diff
@@ -0,0 +1,49 @@
+// eslint-disable-next-line spaced-comment
+/// <reference types="../../global" />
+import Cache from 'sync-disk-cache';
+import os from 'os';
+import { EditingData } from '../sharedTypes/editing-data';
+
+/**
+ * Defines an editing data cache implementation
+ */
+export interface EditingDataCache {
+  set(key: string, editingData: EditingData): void;
+  get(key: string): EditingData | undefined;
+}
+
+/**
+ * A disk-based editing data cache implementation (required for hosting on Vercel via Serverless Functions)
+ * @see EditingDataCache
+ */
+export class EditingDataDiskCache implements EditingDataCache {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private cache: any;
```

Comment:
I'm torn. While I agree 'any' is bad, it seems like overkill just to avoid it for 3rd party libraries that don't ship with or publish (via Definitely Typed) type definitions.

Also, we've already got an `EditingDataCache` interface :)

Reply:
@ambrauer Agree, in such cases we can have exception, we can use `any`

Reply:
@ambrauer Yes...but `EditingDataCache` is different interface, totally. May be can you define separate type for `sync-disk-cache` in `global.d.ts` file?

Reply:
@ambrauer I tested it, you can define separate type in `global.d.ts`:
```ts
// global.d.ts
declare module 'sync-disk-cache' {
  export default import('sync-disk-cache').default;
  export interface CacheInstance {
    root: string;
    set(key: string, editingData: string): string;
    get(key: string): { value: string; isCached?: boolean };
    remove(key: string): void;
  }
}

// editing-data-cache.ts
import Cache, { CacheInstance } from 'sync-disk-cache';

export class EditingDataDiskCache implements EditingDataCache {
  private cache: CacheInstance;

  constructor() {
    // Use gzip compression and store using the OS temp directory (Vercel Serverless Functions have temp directory access)
    this.cache = new Cache('editing-data', { compression: 'gzip', location: os.tmpdir() });
  }
}

```

Reply:
@sc-illiakovalenko I like it. Done.